docs/Upgrade-Guide/Generic_Upgrade_procedure.md

```diff
@@ -0,0 +1,104 @@
+# Generic Upgrade procedure
+
+
+### Pre-upgrade notes
+- Online upgrade is only possible with replicated and distributed replicate volumes
+- Online upgrade is not supported for dispersed or distributed dispersed volumes
+- Ensure no configuration changes are done during the upgrade
+- If you are using geo-replication, please upgrade the slave cluster(s) before upgrading the master
+- Upgrading the servers ahead of the clients is recommended
+- It is recommended to have the same client and server, major versions running eventually
+
+### Online upgrade procedure for servers
+This procedure involves upgrading **one server at a time**, while keeping the volume(s) online and client IO ongoing. This procedure assumes that multiple replicas of a replica set, are not part of the same server in the trusted storage pool.
+
+> **ALERT:** If there are disperse or, pure distributed volumes in the storage pool being upgraded, this procedure is NOT recommended, use the [Offline upgrade procedure](#offline-upgrade-procedure) instead.
+
+#### Repeat the following steps, on each server in the trusted storage pool, to upgrade the entire pool to new-version :
+1. Stop all gluster services, either using the command below, or through other means,
+
+        # killall glusterfs glusterfsd glusterd
```

Comment:
Ideally glusterd should be stopped using systemctl.  so it should be
```
# systemctl stop glusterd
# killall glusterfs glusterfsd
```

Reply:
Done, as discussed will also keep 
# systemctl stop glustereventsd